pkg/utils/predicate.go

```diff
@@ -21,144 +21,305 @@ import (
 	}
 }
 
-// OwnersOwnerInNamespacePredicate filters events to check the owners owner of
-// the event object is in the controller's namespace
-type OwnersOwnerInNamespacePredicate struct {
-	client                    client.Client
-	ownerInNamespacePredicate OwnerInNamespacePredicate
+// Create returns true if the event object owner is a GitTrack
+func (p OwnerIsGitTrackPredicate) Create(e event.CreateEvent) bool {
+	return p.ownerIsGitTrack(e.Meta.GetOwnerReferences())
 }
 
-// Create returns true if the event object owners owner is in the same namespace
-func (p OwnersOwnerInNamespacePredicate) Create(e event.CreateEvent) bool {
-	return p.ownersOwnerInNamespace(e.Meta.GetOwnerReferences())
+// Update returns true if the event object owner is a GitTrack
+func (p OwnerIsGitTrackPredicate) Update(e event.UpdateEvent) bool {
+	return p.ownerIsGitTrack(e.MetaNew.GetOwnerReferences())
 }
 
-// Update returns true if the event object owners owner is in the same namespace
-func (p OwnersOwnerInNamespacePredicate) Update(e event.UpdateEvent) bool {
-	return p.ownersOwnerInNamespace(e.MetaNew.GetOwnerReferences())
+// Delete returns true if the event object owner is a GitTrack
+func (p OwnerIsGitTrackPredicate) Delete(e event.DeleteEvent) bool {
+	return p.ownerIsGitTrack(e.Meta.GetOwnerReferences())
 }
 
-// Delete returns true if the event object owners owner is in the same namespace
-func (p OwnersOwnerInNamespacePredicate) Delete(e event.DeleteEvent) bool {
-	return p.ownersOwnerInNamespace(e.Meta.GetOwnerReferences())
+// Generic returns true if the event object owner is a GitTrack
+func (p OwnerIsGitTrackPredicate) Generic(e event.GenericEvent) bool {
+	return p.ownerIsGitTrack(e.Meta.GetOwnerReferences())
 }
 
-// Generic returns true if the event object owners owner is in the same namespace
-func (p OwnersOwnerInNamespacePredicate) Generic(e event.GenericEvent) bool {
-	return p.ownersOwnerInNamespace(e.Meta.GetOwnerReferences())
+func (p OwnerIsGitTrackPredicate) ownerIsGitTrack(ownerRefs []metav1.OwnerReference) bool {
+	for _, ref := range ownerRefs {
+		if ref.Kind == "GitTrack" && ref.APIVersion == farosGroupVersion {
+			return true
+		}
+	}
+	return false
 }
 
-// ownersOwnerInNamespace returns true if the the GitTrackObject's GitTrack
-// owner of the event object is in the namespace  managed by the controller
-//
-// This works on the premise that listing objects from the client will only
-// return those in its cache.
-// When it is restricted to a namespace this should only be the GitTracks
-// in the namespace the controller is managing.
-func (p OwnersOwnerInNamespacePredicate) ownersOwnerInNamespace(ownerRefs []metav1.OwnerReference) bool {
-	cgtoList := &farosv1alpha1.ClusterGitTrackObjectList{}
-	err := p.client.List(context.TODO(), cgtoList)
-	if err != nil {
-		// We can't list CGTOs so fail closed and ignore the requests
-		return false
+// OwnersOwnerIsGitTrackPredicate filters events to check the owner's owner of the event
```

Comment:
I don't `owner's owner of` makes sense? Consider this instead
```suggestion
// OwnersOwnerIsGitTrackPredicate filters events to check the owner of the event object's owner is a GitTrack
```